### ISSUE TITLE
After upgrading to latest version attempting to play throws an "list index out of range" error

### ISSUE THREAD

**Issue by pabloromeo** [maykar/plex_assistant#92]
Context: 
- Installed through HACS
- Using IFTTT
- Command: "Tell Plex to play Family Guy on the TV"

Given that the latest version had breaking changes, I removed the previous version entirely, removed the config keys and re-added it as an Integration. All went well, and configured successfully against Plex and found the TV as a cast device.
However, when sending a command, it throws the exception:

```
Error executing service: <ServiceCall plex_assistant.command (c:3565d9199c6da779cb526c5a1946481e): command=play Family Guy on the TV>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1471, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1494, in _execute_service
    await self._hass.async_add_executor_job(handler.job.target, service_call)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/plex_assistant/__init__.py", line 193, in handle_input
    media = pa.server.createPlayQueue(media.show().episodes(), startItem=media)
  File "/usr/local/lib/python3.8/site-packages/plexapi/server.py", line 353, in createPlayQueue
    return PlayQueue.create(self, item, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/plexapi/playqueue.py", line 190, in create
    c = cls(server, data, initpath=path)
  File "/usr/local/lib/python3.8/site-packages/plexapi/base.py", line 51, in __init__
    self._loadData(data)
  File "/usr/local/lib/python3.8/site-packages/plexapi/playqueue.py", line 66, in _loadData
    self.selectedItem = self[self.playQueueSelectedItemOffset]
  File "/usr/local/lib/python3.8/site-packages/plexapi/playqueue.py", line 71, in __getitem__
    return self.items[key]
IndexError: list index out of range
```

Any idea why this might be happenning?

**Comment by maykar** [comment]
Can you enable the debug logs as motioned in the troubleshooting docs listed above so we can get more info into what's happening there?

**Comment by Kravatak76** [comment]
I have same error.

**Comment by maykar** [comment]
@Kravatak76 maybe you could also provide the information requested in my previous comment?

**Comment by Kravatak76** [comment]
I can try, but I'm not an expert and I don't know the logos, so I apologize if it's not right. I had no problems with media players before updating and installing the integration. 
[log.txt](https://github.com/maykar/plex_assistant/files/5975704/log.txt)

**Comment by maykar** [comment]
Thinking this might be due to large amount of episodes for a show. Could you both try narrowing it down to see if you get the same error? So, try:

@pabloromeo  `Tell Plex to play season 8 Family Guy on the TV`

@Kravatak76 `Tell Plex to play season 8 of the Simpsons on the on bed TV`

**Comment by pabloromeo** [comment]
@maykar it seems you are correct about it working when you specify a season. 
Specifying the Season does indeed start playback, and it doesn't seem to have issues with movies either.

Here are the logs of when using the more generic command:

```
2021-02-13 11:42:09 DEBUG (SyncWorker_22) [custom_components.plex_assistant.const] IFTTT Call: play Family Guy on the TV
2021-02-13 11:42:09 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=plex_assistant, service=command, service_data=command=play Family Guy on the TV>
2021-02-13 11:42:09 DEBUG (SyncWorker_11) [custom_components.plex_assistant.const] Command: play Family Guy on the TV
2021-02-13 11:42:10 DEBUG (SyncWorker_11) [custom_components.plex_assistant.const] Processed Command: media = family guy, device = tv
2021-02-13 11:42:10 DEBUG (SyncWorker_11) [custom_components.plex_assistant.const] Device: TV
2021-02-13 11:42:11 DEBUG (SyncWorker_11) [custom_components.plex_assistant.const] Media: <Episode:5729:Family-Guy-s09e16>
2021-02-13 11:42:15 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall plex_assistant.command (c:d8bb96d514cc0a328021172de20e44b2): command=play Family Guy on the TV>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1471, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1494, in _execute_service
    await self._hass.async_add_executor_job(handler.job.target, service_call)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/plex_assistant/__init__.py", line 193, in handle_input
    media = pa.server.createPlayQueue(media.show().episodes(), startItem=media)
  File "/usr/local/lib/python3.8/site-packages/plexapi/server.py", line 353, in createPlayQueue
    return PlayQueue.create(self, item, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/plexapi/playqueue.py", line 190, in create
    c = cls(server, data, initpath=path)
  File "/usr/local/lib/python3.8/site-packages/plexapi/base.py", line 51, in __init__
    self._loadData(data)
  File "/usr/local/lib/python3.8/site-packages/plexapi/playqueue.py", line 66, in _loadData
    self.selectedItem = self[self.playQueueSelectedItemOffset]
  File "/usr/local/lib/python3.8/site-packages/plexapi/playqueue.py", line 71, in __getitem__
    return self.items[key]
IndexError: list index out of range
```

It does appear to pick up an episode for playback (Episode:5729:Family-Guy-s09e16), but the command for the playqueue fails, it seems.

**Comment by maykar** [comment]
Cool, so it seems there is an undocumented limit to how many items can be returned in a playQueue or another limiting factor. It would be useful to know what that limit is. My entire library isn't big enough to trigger this, so if either of you are willing to test this out for me:

On line 192 of `__init__.py` find:

```
        if getattr(media, "TYPE", None) == "episode":
            media = pa.server.createPlayQueue(media.show().episodes(), startItem=media)
```

And replace it with:

```
        if getattr(media, "TYPE", None) == "episode":
            _LOGGER.warning("Total Episodes: %s", len(media.show().episodes()))
            test = pa.server.createPlayQueue(media.show().episodes())
            _LOGGER.warning("PlayQueue Size: %s", test.size)
            media = pa.server.createPlayQueue(media.show().episodes(), startItem=media, shuffle=shuffle)
```

Restart and use the original command you were using: "Tell Plex to play Family Guy on the TV"

In your logs you'll have 2 items I'm interested in "Total Episodes" and "PlayQueue Size"

**Comment by Kravatak76** [comment]
2021-02-14 19:30:43 WARNING (SyncWorker_5) [custom_components.plex_assistant.const] Total Episodes: 695
2021-02-14 19:30:53 WARNING (SyncWorker_5) [custom_components.plex_assistant.const] PlayQueue Size: 21

**Comment by maykar** [comment]
That's quite a bit lower than I expected. Thanks, will update accordingly.

**Comment by maykar** [comment]
Tracked this one down and it actually doesn't have to do with that playQueue size from above, seems in debugging this we've found an interesting bug in Plex's API when it comes to playQueue items. Will have a fix in the next version.

**Comment by maykar** [comment]
Fix in version 1.0.3

**Comment by Kravatak76** [comment]
I upgraded to version 1.0.3 and the error no longer occurs. thank you for the quick fix
